### DISCUSSION TITLE
DearImGui support

### DISCUSSION THREAD

**Discussion by pbk0** [szabolcsdombi/zengl#56] [Ideas]
Thanks for zengl, especially the cool pyscript demos you released recently. 
Really wanted something like moderngl (your earlier project) on the web.
I hope DearImGui is on your hit list. This will make us port our code directly on web browsers.

**Comment by szabolcsdombi** [comment]
Hello,
Glad to hear you find zengl useful,

We can render DearImGui, see https://github.com/szabolcsdombi/zengl/issues/50
The problem at is that DearImGui requires user events and this part is way more complex than just rendering the imgui provided index/vertex data.

For this reason imgui needs its own web port. which already exists.

I think you can use it just like any other javascript library from PyScript.

I think rendering imgui with zengl adds no value. The proper way should be to render the scene with zengl

```py
ctx.new_frame()
...
ctx.end_frame()
# after a zengl frame you can invoke the web version of imgui directly from pyscript.
```

and render imgui with a javascript version of it at the end.
